source/Halibut/Transport/RewindableBufferStream.cs

```diff
@@ -114,7 +114,7 @@ public async override Task<int> ReadAsync(byte[] buffer, int offset, int count,
                 return rewoundCount;
             }
 
-            var baseCount = await baseStream.ReadAsync(buffer, offset, count);
+            var baseCount = await baseStream.ReadAsync(buffer, offset, count, cancellationToken);
```

Comment:
Are there any tests that currently exist or we could potentially add that would help us to assert the expected behaviour around cancellation? E.g. Start a stream, cancel the read via the token and make sure it doesn't hang?

Reply:
I think only one test exists in halibut that checks timeouts work but that goes down a sync path. I can take a closer look tomorrow. It is not clear to me if this async read method is ever called currently.

Reply:
> It is not clear to me if this async read method is ever called currently.

If we aren't sure if it's called, then are we sure this change will fix the underlying issue? Feels like it would be good to get some confidence in the change through some tests.

Reply:
Testing the read timeout works when the client waits for the service to respond is difficult to write currently, since the read timeout comes from the static
```
public static TimeSpan TcpClientReceiveTimeout = TimeSpan.FromMinutes(10);
```
and so the test would take ten minutes to run.

I guess that could be changed to take the timeout from the ServiceEndpoint.

I guess a simpler test is one that targets the RewindableBufferStream but that doesn't really capture the behaviour and doesn't show the timeout working when waiting for the service to respond.

In any case passing the cancellation token along makes sense.

Reply:
@geofflamrock I created a targeted test for this.